### PR TITLE
[WIP] ENH CUDA 11.0 support

### PR DESCRIPTION
**Pull request by mike-wendt**

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -112,12 +111,11 @@ outputs:
     requirements:
       build:
         - {{ compiler("c") }}
+        - sysroot_linux-64 2.17  # [linux64]
```
Comment (jakirkham) [inline recipe/meta.yaml:114]:
```suggestion
        - sysroot_linux-64 2.17  # [linux64 and cuda_compiler_version == "11.0"]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -69,12 +69,11 @@ outputs:
...
```
Comment (jakirkham) [inline recipe/meta.yaml:72]:
```suggestion
        - sysroot_linux-64 2.17  # [linux64 and cuda_compiler_version == "11.0"]
```

**Comment by jakirkham** [comment]
Re-rendered locally and manually kept the CUDA 11 changes, which gives commit ( https://github.com/jakirkham-feedstocks/ucx-split-feedstock/commit/14598a46bd80dd1c13e7d2bb541649152d2bca1a ). Could we try adding that here?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -72,7 +72,7 @@ outputs:
         - sysroot_linux-64 2.17 # [linux64]
         - {{ compiler("cxx") }}
         - {{ compiler("cuda") }}         # [cuda_compiler_version != "None"]
-        - {{ cdt("libnl") }}
+        - {{ cdt("libnl3") }}
```
Comment (jakirkham) [inline recipe/meta.yaml:75]: Based on what is seen in [this log]( https://gpuci.gpuopenanalytics.com/job/rapidsai/job/conda/job/ucx/job/ucx-py-nightly-cuda110-take2/lastBuild/CONDA_USERNAME=rapidsai-nightly,CUDA_VER=11.0,FROM_IMAGE=mikewendt%2Flinux-anvil-cuda,UCX_COMMIT=v1.8.x,UCX_PROC_VER=1.0.0,UCX_PY_COMMIT=branch-0.15,UCX_VER=1.8.0dev/consoleFull ) in particular lines like the one below, it looks like we need `libnl3`.

Tried to add this with PR ( https://github.com/conda-forge/cdt-builds/pull/26 ), but had some trouble getting the CDT generated. So am trying to add these manually in PR ( https://github.com/conda-forge/cdt-builds/pull/28 ). That should fix this issue.

```
.../libibverbs.so: undefined reference to `nl_cache_mngt_unprovide@libnl_3'
```

**Comment by jakirkham** [comment]
Looking at [the most recent build log]( https://gpuci.gpuopenanalytics.com/job/rapidsai/job/conda/job/ucx/job/ucx-py-nightly-cuda110-take2/lastBuild/CONDA_USERNAME=rapidsai-nightly,CUDA_VER=11.0,FROM_IMAGE=mikewendt%2Flinux-anvil-cuda,UCX_COMMIT=v1.8.x,UCX_PROC_VER=1.0.0,UCX_PY_COMMIT=branch-0.15,UCX_VER=1.8.0dev/consoleFull ), I see the following, which shows `uct/ib/rc` being built. However I don't see that transport included in the final package, which is kind of interesting.

```
  CC       rc/base/libuct_ib_la-rc_ep.lo
  CC       rc/base/libuct_ib_la-rc_iface.lo
  CC       rc/verbs/libuct_ib_la-rc_verbs_ep.lo
  CC       rc/verbs/libuct_ib_la-rc_verbs_iface.lo
  CC       rc/accel/libuct_ib_la-rc_mlx5_ep.lo
  CC       rc/accel/libuct_ib_la-rc_mlx5_iface.lo
  CC       rc/accel/libuct_ib_la-rc_mlx5_common.lo
  CC       rc/accel/libuct_ib_la-rc_mlx5_devx.lo
```

cc @pentschev

**Comment by pentschev** [comment]
Yes, I think that's normal, they're not installed when building from source on baremetal either:

```
$ find /datasets/pentschev/miniconda3/envs/ucx-src-110-0.15.0b200730 -iname "libuct_ib*"
/datasets/pentschev/miniconda3/envs/ucx-src-110-0.15.0b200730/lib/ucx/libuct_ib.so.0.0.0
/datasets/pentschev/miniconda3/envs/ucx-src-110-0.15.0b200730/lib/ucx/libuct_ib.so.0
/datasets/pentschev/miniconda3/envs/ucx-src-110-0.15.0b200730/lib/ucx/libuct_ib.so
/datasets/pentschev/miniconda3/envs/ucx-src-110-0.15.0b200730/lib/ucx/libuct_ib.la
/datasets/pentschev/miniconda3/envs/ucx-src-110-0.15.0b200730/lib/ucx/libuct_ib.a
/datasets/pentschev/miniconda3/envs/ucx-src-110-0.15.0b200730/lib/ucx/libuct_ib_cm.so.0.0.0
/datasets/pentschev/miniconda3/envs/ucx-src-110-0.15.0b200730/lib/ucx/libuct_ib_cm.so.0
/datasets/pentschev/miniconda3/envs/ucx-src-110-0.15.0b200730/lib/ucx/libuct_ib_cm.so
/datasets/pentschev/miniconda3/envs/ucx-src-110-0.15.0b200730/lib/ucx/libuct_ib_cm.la
/datasets/pentschev/miniconda3/envs/ucx-src-110-0.15.0b200730/lib/ucx/libuct_ib_cm.a
```

**Comment by jakirkham** [comment]
Seems NUMA is disabled. Maybe we don't have the right CDTs here

```
configure: NUMA support is disabled
```

**Comment by mike-wendt** [comment]
Closing in favor of a new PR that pulls in changes from #40 and changes here

**Comment by jakirkham** [comment]
Handled in PR ( https://github.com/rapidsai/ucx-split-feedstock/pull/41 ).

**Pull request closed without merging**